src/main/java/org/opensearch/ad/AnomalyDetectorJobRunner.java

```diff
@@ -528,25 +531,25 @@ private void updateRealtimeTask(AnomalyResultResponse response, String detectorI
             ProfileRequest profileRequest = new ProfileRequest(detectorId, profiles, true, dataNodes);
             client.execute(ProfileAction.INSTANCE, profileRequest, ActionListener.wrap(r -> {
                 log.debug("Update latest realtime task for HC detector {}, total updates: {}", detectorId, r.getTotalUpdates());
-                adTaskManager
-                    .updateLatestRealtimeTask(
-                        detectorId,
-                        null,
-                        r.getTotalUpdates(),
-                        response.getDetectorIntervalInMinutes(),
-                        response.getError()
-                    );
-            }, e -> { log.error("Failed to get init progress profile for " + detectorId, e); }));
-        } else {
-            log.debug("Update latest realtime task for SINGLE detector {}, total updates: {}", detectorId, response.getRcfTotalUpdates());
-            adTaskManager
-                .updateLatestRealtimeTask(
+                updateLatestRealtimeTask(
+                    jobParameter,
                     detectorId,
                     null,
-                    response.getRcfTotalUpdates(),
+                    r.getTotalUpdates(),
                     response.getDetectorIntervalInMinutes(),
                     response.getError()
                 );
+            }, e -> { log.error("Failed to get init progress profile for " + detectorId, e); }));
```

Comment:
error message doesn't make sense here? Maybe change to something like `Failed to update latest realtime task for " + detectorId, e`

Reply:
good catch, will fix